core/src/main/java/org/apache/iceberg/avro/TypeToSchema.java

```diff
@@ -214,4 +216,58 @@ public Schema primitive(Type.PrimitiveType primitive) {
 
     return primitiveSchema;
   }
+
+  private static String sanitize(String name) {
+    if (validAvroName(name)) {
+      return name;
+    }
+    int length = name.length();
+    StringBuilder sb = new StringBuilder(name.length());
+    char first = name.charAt(0);
+    if (!(Character.isLetter(first) || first == '_')) {
+      sb.append(sanitizedValue(first));
+    } else {
+      sb.append(first);
+    }
+    for (int i = 1; i < length; i++) {
+      char character = name.charAt(i);
+      if (!(Character.isLetterOrDigit(character) || character == '_')) {
+        sb.append(sanitizedValue(character));
+      } else {
+        sb.append(character);
+      }
+    }
+    return sb.toString();
+  }
+
+  private static final Map<Character, String> SANITIZE_MAP = new ImmutableMap
```

Comment:
Instead of using a map and checking whether a special character is supported, why not just convert any non-letter value to its unicode hex representation, like `_x2E` for `.`? That way, all symbols are supported.

Reply:
Yea, that seems like a good idea. I'll try that.

Reply:
By the way, this should include a test for a multi-byte character.

Reply:
ofcourse :)